### PR TITLE
feat(js-loader): Make latest unselectable

### PR DESCRIPTION
**Pull request by lforst**
We want to remove the option to select "latest" in the JS loader because it makes the maintenance of the JS SDK hell in terms of backwards compatibility.

For this we need to start mapping "latest", when selected, to the actual latest value, which at the current time is "7.x".

We are starting in the frontend:
- When the backend says "latest" is a valid option to select, we remove that option from the list of selectable items.
- Because of the above, we also need to map the "latest" value, if the backend says it is selected, to "7.x" so that the select knows what selection to show because "latest" is not an option anymore.
- Previously, there was never the case where only one item was selectable. After this change, there are situations where only "7.x" may be selected. In this case we disable the select (because there are no more valid options) and show a hint explaining that only one option can be selected and why.

In a follow-up, we will update the backend to limit what options are returned. (follow up: https://github.com/getsentry/sentry/pull/65691 - both could in theory be deployed independently but having this PR out first is a tiny bit better because of the disabled state of the select)

### CODE REVIEW

File: static/app/views/settings/project/projectKeys/details/loaderSettings.tsx [hunk 1]
```diff
@@ -40,12 +40,21 @@ export function LoaderSettings({keyId, orgSlug, project, data, updateData}: Prop
   const values = requestPending
     ? optimisticState
     : {
-        browserSdkVersion: data.browserSdkVersion,
+        browserSdkVersion:
+          // "latest" was an option that we do not users to select anymore. It will be phased out when version v8 of
+          // the SDK is released, meaning we want to map the backends response to v7 when it responds with "latest".
```
Comment (Lms24) [inline static/app/views/settings/project/projectKeys/details/loaderSettings.tsx:45]: nit/feel free to ignore:

```suggestion
          // the SDK is released, meaning we want to map the backend's response to v7 when it responds with "latest".
```

**Review by Lms24 — APPROVED** [review]
Nice! Just a couple of remarks.

File: static/app/views/settings/project/projectKeys/details/loaderSettings.tsx [hunk 2]
```diff
@@ -148,21 +157,24 @@ export function LoaderSettings({keyId, orgSlug, project, data, updateData}: Prop
...
             value={values.browserSdkVersion}
             onChange={value => {
               updateLoaderOption({browserSdkVersion: value});
             }}
+            disabledReason={
+              sdkVersionChoices.length === 1
+                ? t(
+                    'At the moment, only the shown SDK version is available. New versions of the SDK will appear here as soon as they are released, and you will be able to upgrade by selecting them.'
+                  )
+                : undefined
```
[the remark this answers is not in the document]
Comment (lforst) [reply static/app/views/settings/project/projectKeys/details/loaderSettings.tsx:173]: Recently created projects will not be shown any versions prior to v7. This relates to: https://github.com/getsentry/sentry/pull/65691#discussion_r1500585164

[the remark this answers is not in the document]
Comment (Lms24) [reply static/app/views/settings/project/projectKeys/details/loaderSettings.tsx:173]: Ah you already explained this with the epochs here: https://github.com/getsentry/sentry/pull/65691#discussion_r1500585164; thanks!

**Pull request merged**